ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/baseActivity.component.ts

```diff
@@ -154,6 +155,20 @@ export abstract class BaseActivityComponent implements OnChanges, OnDestroy {
         this.activityGuidObservable.next(this.activityGuidObservable.value);
     }
 
+    public updateVisibility(visibility: BlockVisibility[]): void {
+        visibility.forEach(element => {
+            this.model.sections.forEach(section => {
+                section.blocks.forEach(block => {
+                    if (block.id === element.blockGuid) {
+                        block.shown = element.shown;
+                    }
```

Comment:
I tried this fix locally and seems to work for studies like brain/osteo/RGP. However, I feel a bit uneasy about adding back this code and worry about how it interacts with similar code in `ActivitySectionComponent`.

What if we made a bit more refactor here, what do folks think about this additional change?
```diff
diff --git a/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activitySection.component.ts b/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activitySection.component.ts
index cbd77648..89a39691 100644
--- a/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activitySection.component.ts
+++ b/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activitySection.component.ts
@@ -70,10 +70,9 @@ export class ActivitySectionComponent implements OnInit, OnDestroy {
                     if (block.shown !== element.shown) {
                         block.shown = element.shown;
                         blockVisibilityChanged = true;
-
-                        if (block.blockType === BlockType.Activity && !block.shown) {
-                            this.updateValidationForHiddenEmbeddedActivity(block as ActivityActivityBlock);
-                        }
+                    }
+                    if (block.blockType === BlockType.Activity && !block.shown) {
+                        this.updateValidationForHiddenEmbeddedActivity(block as ActivityActivityBlock);
                     }
                 }
             });
```

Idea is we pull out the `if (BlockType.Activity)` check one level up, so we always run it and ensure the hidden embedded activities state tracking is up-to-date for those.

Reply:
Fixed